Code/Engine/Core/World/Implementation/World.cpp

```diff
@@ -995,12 +995,13 @@ void ezWorld::UpdateAsynchronous()
     if (updateFunction.m_bOnlyUpdateWhenSimulating && !m_Data.m_bSimulateWorld)
       continue;
 
-    ezComponentManagerBase* pManager = static_cast<ezComponentManagerBase*>(updateFunction.m_Function.GetClassInstance());
+    ezWorldModule* pModule = static_cast<ezWorldModule*>(updateFunction.m_Function.GetClassInstance());
+    ezComponentManagerBase* pManager = ezDynamicCast<ezComponentManagerBase*>(pModule);
 
-    const ezUInt32 uiTotalCount = pManager->GetComponentCount();
-    ezUInt32 uiStartIndex = 0;
-    ezUInt32 uiGranularity = (updateFunction.m_uiGranularity != 0) ? updateFunction.m_uiGranularity : uiTotalCount;
+    const ezUInt32 uiTotalCount = pManager != nullptr ? pManager->GetComponentCount() : 1;
+    const ezUInt32 uiGranularity = (updateFunction.m_uiGranularity != 0) ? updateFunction.m_uiGranularity : uiTotalCount;
 
+    ezUInt32 uiStartIndex = 0;
```

Comment:
Is this just code cleanup, or is there a functional change?

Reply:
This is a fix, that code crashed if an ezWorldModules (not a component manager) registered an update function for the async phase.